donkeycar/parts/realsense435i.py

```diff
@@ -0,0 +1,301 @@
+"""
+Author: Ed Murphy
+File: realsense435i.py
+Date: April 14 2019
+Notes: Donkeycar part for the Intel Realsense depth cameras D435 and D435i.
+"""
+import time
+import logging
+
+import numpy as np
+import pyrealsense2 as rs
+
+#
+# NOTE: Jetson Nano users should clone the Jetson Hacks project
+#       https://github.com/JetsonHacksNano/installLibrealsense
+#       and _build librealsense from source_ in order to get the python bindings.
+#       ./buildLibrealsense.sh
+#
+
+#
+# The Realsense D435 will not output images with arbitrarily chosen dimensions.
+# The dimensions must be from the allowed list.  I've chosen a reasonable
+# resolution that I know is supported by the camera.
+# If the part is initialized with different sizes, then opencv
+# will be used to resize the image before it is returned.
+#
+WIDTH = 424
+HEIGHT = 240
+CHANNELS = 3
+
+class RealSense435i(object):
+    """
+    Donkeycar part for the Intel Realsense depth cameras D435 and D435i.
+    The Intel Realsense D435i camera is a device which uses an imu, twin fisheye cameras,
+    and an Movidius chip to stream a depth map along with an rgb image and optionally,
+    accelerometer and gyro data (the 'i' variant has an IMU, the non-i variant does not)
+    NOTE: this ALWAYS captures 424 pixels wide x 240 pixels high x RGB at 60fps.
+          If an image width, height or depth are passed with different values,
+          the the outputs will be scaled to the requested size on the way out.
+    """
+
+    def __init__(self, width = WIDTH, height = HEIGHT, channels = CHANNELS, enable_rgb=True, enable_depth=True, enable_imu=False, device_id = None):
+        self.device_id = device_id  # "923322071108" # serial number of device to use or None to use default
+        self.enable_imu = enable_imu
+        self.enable_rgb = enable_rgb
+        self.enable_depth = enable_depth
+
+        self.width = width
+        self.height = height
+        self.channels = channels
+        self.resize = (width != WIDTH) or (height != height) or (channels != CHANNELS)
+        if self.resize:
+            print("The output images will be resized from {} to {}.  This requires opencv.".format((WIDTH, HEIGHT, CHANNELS), (self.width, self.height, self.channels)))
+
+        # Configure streams
+        self.imu_pipeline = None
+        if self.enable_imu:
+            self.imu_pipeline = rs.pipeline()
```

Comment:
Do you need a pipeline per feature ?

Reply:
Not a pipeline per feature, but IMU does _have_ to be a separate pipeline.  It doesn't work otherwise.